src/foam/dao/PostgresDAO.java

```diff
@@ -0,0 +1,175 @@
+package foam.dao;
+
+import foam.core.*;
+
+import java.sql.*;
+
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.List;
+
+public class PostgresDAO extends ProxyDAO {
+
+    public PostgresDAO(DAO delegate, String host, String port, String dbName, String username, String password) {
+        setDelegate(delegate);
+
+        if (dbName == null || username == null) {
+            throw new IllegalArgumentException("Illegal arguments");
+        }
+
+        host = (host != null) ? host : "localhost";
+        port = (port != null) ? port : "5432";
+
+        PgConnectionPool.setup(host, port, dbName, username, password);
+    }
+
+    @Override
+    public FObject remove(FObject o) {
+
+        try {
+
+            SQLData data = new SQLData(o);
+            Connection c = PgConnectionPool.getConnection();
+
+            PreparedStatement smt = c.prepareStatement(data.createDeleteStatement());
+
+            smt.setLong(1, ((Long) o.getProperty("id")));
```

Comment:
We don't know that the 'id' property is a Long. (Which is okay for now, but at least add a TODO.)